src/features/validation/validate.ts

```diff
@@ -208,8 +208,24 @@ export class ApplyMarkersCommand extends FeedbackCommand {
     }
 }
 
-function addCSSClassToIssueParent(modelElement: SParentElement, issueMarker: SIssueMarker) {
-    addCssClasses(modelElement, [getSeverity(issueMarker)]);
+function addMaxSeverityCSSClassToIssueParent(modelElement: SParentElement, issueMarker: SIssueMarker) {
+    const severityCSSClass = getSeverity(issueMarker);
+
+    if (modelElement.cssClasses === undefined) {
+        modelElement.cssClasses = [severityCSSClass];
+    } else {
+        modelElement.cssClasses.push(severityCSSClass);
+        modelElement.cssClasses = getMaxSeverity(modelElement.cssClasses);
```

Comment:
Doesn't this remove all other cssClasses that might be attached to the modelElement/issueMarker ?

Reply:
Thanks, you're right! I edited this part, now it makes sure that only severity cssClasses are replaced by the max severity cssClass of the parent modelElement.